app/src/main/java/org/simple/clinic/activity/TheActivity.kt

```diff
@@ -96,10 +103,10 @@ class TheActivity : AppCompatActivity() {
   }
 
   private fun initialScreenKey(): FullScreenKey {
-    return if (userSession.isUserLoggedIn()) {
-      HomeScreen.KEY
-    } else {
-      RegistrationPhoneScreen.KEY
+    return when {
+      hasUserCompletedOnboarding.get().not() -> OnboardingScreen.KEY
+      userSession.isUserLoggedIn() -> HomeScreen.KEY
+      else -> RegistrationPhoneScreen.KEY
     }
```

Comment:
Good point. Except the nested `when` doesn't look very good since there are only two cases, so I think I'll flatten them out like this.

```kotlin
when {
  userSession.isLoggedIn() -> HomeScreen.KEY
  hasUserCompletedOnboarding.get().not() -> OnboardingScreen.KEY
  else -> RegistrationPhoneScreen.KEY
}
```

Reply:
👍